File: frontend/src/services/report.service.ts
import axiosInstance from '../config/axios.instance';
import { ApiResponse } from '../types/api';
import { ClassAttendanceReportResponse } from '../types/report.type';

export const getClassAttendanceReport = async (classId: number, month: number, year: number): Promise<ApiResponse<ClassAttendanceReportResponse>> => {
  const response = await axiosInstance.get(`/api/core/reports/attendance`, {
    params: { classId, month, year }
  });
  return response.data;
};

File: preschool-management-backend/src/main/java/com/vusystem/preschool_management_backend/modules/core/services/ReportService.java
package com.vusystem.preschool_management_backend.modules.core.services;

import com.vusystem.preschool_management_backend.modules.core.dto.response.ClassAttendanceReportResponse;

public interface ReportService {
    ClassAttendanceReportResponse generateClassAttendanceReport(Long classId, int month, int year);
}
